packages/sdk/src/oracle/services/statemachines/statemachine.ts

```diff
@@ -284,6 +292,13 @@ export class StateMachine {
           );
           break;
         }
+        case ContextType.fetchEventBased: {
+          next = await this.types[context.type].step(
+            (context as unknown) as Context<fetchEventBased.Params, fetchEventBased.Memory>,
```

Comment:
nit: nothing to change in this PR, but I think you could link the context type and the parameter types so this casting could be inferred by ts.

Basically, you could make each context variant a tagged union that derives from the common base interface. Then once you resolve the type, the types of the other params are inferred.